core/sound/channel3.ts

```diff
@@ -186,46 +179,42 @@ export class Channel3 {
     let sample = 0;
 
     // Will Find the position, and knock off any remainder
-    let positionIndexToAdd = i32Portable(Channel3.waveTablePosition >> 1);
+    let waveTablePosition = Channel3.waveTablePosition;
+    let positionIndexToAdd = i32Portable(waveTablePosition >> 1);
     let memoryLocationWaveSample = Channel3.memoryLocationWaveTable + positionIndexToAdd;
 
     sample = eightBitLoadFromGBMemory(memoryLocationWaveSample);
 
     // Need to grab the top or lower half for the correct sample
-    if ((Channel3.waveTablePosition & 1) === 0) {
-      // First sample
-      sample = sample >> 4;
-      sample = sample & 0x0f;
-    } else {
-      // Second Samples
-      sample = sample & 0x0f;
-    }
+    sample >>= (<i32>((waveTablePosition & 1) === 0)) << 2;
+    sample &= 0x0f;
 
     // Shift our sample and set our volume depending on the volume code
     // Since we can't multiply by float, simply divide by 4, 2, 1
     // http://gbdev.gg8.se/wiki/articles/Gameboy_sound_hardware#Wave_Channel
+    let outputVolume = 0;
     switch (volumeCode) {
       case 0:
-        sample = sample >> 4;
+        sample >>= 4;
         break;
       case 1:
         // Dont Shift sample
         outputVolume = 1;
         break;
       case 2:
-        sample = sample >> 1;
+        sample >>= 1;
         outputVolume = 2;
         break;
       default:
-        sample = sample >> 2;
+        sample >>= 2;
         outputVolume = 4;
         break;
     }
 
     // Spply out output volume
     sample = outputVolume > 0 ? sample / outputVolume : 0;
```

Comment:
This pretty strange part. In switch block you first shift `sample` value (actually div by `16`, `2` and `4`) and also return divider `outputVolume` which later divide `sample` by `outputVolume`. If this not mistake you could do all of this in switch blocks like:
```ts
    switch (volumeCode) {
      case 1: break;
      case 2:
        sample >>= 2;
        break;
      case 0:
      default:
        sample >>= 4;
        break;
    }
```
or even simpler:
```ts
if (sample >= 2) {
  sample >>= (volumeCode == 2 ? 2 : 4);
}
```
and that's it

Reply:
Ah that's a good observation! I opened #271 😄